src/Microsoft.AspNetCore.WebHooks.Api/Controllers/WebHookRegistrationsController.cs

```diff
@@ -0,0 +1,315 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.WebHooks;
+using Microsoft.AspNetCore.WebHooks.Api.Routes;
+using Microsoft.AspNetCore.WebHooks.Api.Filters;
+using Microsoft.Extensions.Logging;
+using System.Globalization;
+
+// For more information on enabling Web API for empty projects, visit http://go.microsoft.com/fwlink/?LinkID=397860
+
+namespace Microsoft.AspNetCore.WebHooks.Api.Controllers
+{
+    /// <summary>
+    /// The <see cref="WebHookRegistrationsController"/> allows the caller to create, modify, and manage WebHooks
+    /// through a REST-style interface.
+    /// </summary>
+    [Route("api/webhooks/registrations")]
+    public class WebHookRegistrationsController : Controller
+    {
+        private IWebHookManager _manager;
+        private IWebHookStore _store;
+        private IWebHookUser _user;
+        private ILogger _logger;
+
+        /// <summary>
+        /// Creates a new <see cref="WebHookRegistrationsController"/> which handles management of webhooks
+        /// </summary>
+        /// <param name="manager">The <see cref="IWebHookManager"/> to use</param>
+        /// <param name="store">The <see cref="IWebHookStore"/> to use</param>
+        /// <param name="user">The <see cref="IWebHookUser"/> for the current user</param>
+        /// <param name="loggerFactory">the default <see cref="ILoggerFactory"/> logger factory</param>
+        public WebHookRegistrationsController(IWebHookManager manager,
+            IWebHookStore store, IWebHookUser user, ILoggerFactory loggerFactory)

```

Comment:
BTW you can just take in `ILogger<WebHookRegistrationsController>` and the DI system will provide the correct instance.